internal/daemon/vendor-specific-plugins/intel-netsec/main.go

```diff
@@ -302,20 +429,40 @@ func (vsp *intelNetSecVspServer) setVlanIds(numVfs int) error {
 			return err
 		}
 
+		// Disable spoof check for the VF to allow forwarding traffic.
+		err = vspnetutils.LinkSetVfSpoofChkByName(ifName, vfID, false)
+		if err != nil {
+			vsp.log.Error(err, "Error disabling spoof check for VF", "VFID", vfID, "InterfaceName", ifName)
+			return err
+		}
+
+		// Enable trust for the VF to allow forwarding traffic.
+		err = vspnetutils.LinkSetVfTrustByName(ifName, vfID, true)
```

Comment:
note that we now have these ByName() helpers. They all basically lookup the link instance, and then call a netlink function on them.

makes me wonder, whether those helpers are really useful? Performance-wise now, but that doesn't matter here. But with these helper it also ends up being more code, then just calling `netlink.LinkByName()` once and use it with `netlink.LinkSet*()`. Are they really beneficial?

You talked about stale link entries, but if you lookup the link entre at the beginning of the loop and then use it 3 times, then there is no problem.

Reply:
I decided to change it.